enhancements/machine-config/mco-cgroupsv2-support.md

```diff
@@ -201,19 +202,39 @@ The following jobs will be run against cgroup v2 periodically and with a minimum
 
 ### Upgrade / Downgrade Strategy
 
-Downgrading a cluster to an OpenShift version not containing cgroup v2 support
+- Downgrading a cluster to an OpenShift version not containing cgroup v2 support
 is unsupported.
-
+- Upgrading a cluster which is on cgroup v1 to a version >= 4.19 is blocked until it is migrated to cgroup v2.
+In such case, the MCO cluster operator's status condition is set to `Upgradeable=False` blocking the upgrade.
+- A user/admin has to manually set the `CgroupMode` from `v1` to `v2` as follows to make it `Upgradeable=True` again.
+```shell
+# fetch the nodes.config.openshift.io object
+oc edit nodes.config.openshift.io cluster
+# Update the `CgroupMode` field inside the spec to "v2"
+# Wait for the MCO to rollout cgroup v2 related kernel arguments on all the machine config pools
+```
 ### Version Skew Strategy
 
 A cluster installed with cgroup v2 will abide by the usual skew upgrade path.
 
 #### Removing a deprecated feature
+cgroup v1 support would be removed from the future versions of RHEL and hence the setting of `CgroupMode_V1` would be removed from OCP clusters(>= 4.19)
 
-N/A
+Note: Even after removing the support to configure cgroup v1 from OCP, the `CgroupMode_V1` is not deleted as one of the enum values for the `CgroupMode` in the interest of non-breaking API changes.
```

Comment:
Just to clarify, are we adding CEL validation to reject setting cgroupsv1 entirely at the API level?

Reply:
We intend to reject at the API level using an admission webhook

Reply:
thinking about it more, I think it's actually more idiomatic to use cel validation now. Though ,it does seem weird to me to update the API to add cel validation but leave the v1 field, but I suppose that's what we need to do. cc @rphillips

Reply:
Would this cel validation be at MCO level? What I mean is something similar to [this](https://github.com/openshift/machine-config-operator/blob/master/manifests/machineconfigcontroller/machineconfiguration-guards-validatingadmissionpolicy.yaml)

Reply:
yeah I think so

Reply:
Makes sense to me Peter.
But, wouldn't that be late as @rphillips mentioned that the change would already enter the system if it reaches MCO?
Or is it that the cel validation manifest file gets applied during the bootstrap phase and should be okay?
cc: @yuqi-zhang

Reply:
yeah I think cel happens at a similar time as admission webhooks. My understanding is that cel expression would be registered when the crd is so any user creating a node config object would also interact with the corresponding expression. I may be wrong about details there though

Reply:
@rphillips WDYT about the cel validation ? I would like to update the final draft based on this discussion's output.

Reply:
I'm also not too familiar with the timing differences, but based on my loose understanding both can achieve the result we would want (i.e. have the rejection happen when a user attempts to apply/change the field, instead of retroactively having a controller process the config and error).

CEL validation on the API should be present alongside the CRD, so if the object is able to be created, the validation should always be in place. The MCO wouldn't be rejecting the object in this case, if that's what you're asking

Reply:
+1 Cel validation should work.

Reply:
Updated